CMakeLists.txt

```diff
@@ -53,9 +53,12 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 # Synthesize clang-resource-headers target if necessary.
 if (NOT TARGET clang-resource-headers)
+  execute_process (COMMAND clang -print-resource-dir
```

Comment:
`find_program(CLANG_EXECUTABLE NAMES clang16 clang17 clang REQUIRED)` should be used here for searching for supported Clang versions.

Reply:
@xvitaly using `find_program()` is a great suggestion! But I have some concerns with the parameters being used:

1. This list would have to be reviewed and updated frequently in order to update the supported versions.
2. I think we need to support both formats `clang17` and `clang-17`.
3. I don't think we want to prioritize clang16 over other versions.
4. We should use the same version found previously with `find_package()`.

Please let me know what you think about this implementation I'm proposing.

Reply:
Looks great now. Thank you.